src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy

```diff
@@ -745,57 +742,85 @@ abstract class AbstractServerTask extends AbstractTask {
             if (serverEnvPath != null) {
-                logger.warn("The " + serverEnvPath + " file is merged with the " + server.serverEnvFile.getCanonicalPath() + " file.")
+                logger.debug("The serverEnvFile "+ server.serverEnvFile.getCanonicalPath() + " is merged with the " + serverEnvPath + " file.")
+            }
+            else {
+                logger.debug("The serverEnvFile "+ server.serverEnvFile.getCanonicalPath() + " is merged with the " + getServerDir(project).getCanonicalPath() + " file.")
             }
             Properties configuredServerEnvProps = convertServerEnvToProperties(server.serverEnvFile);
             //Merge with either default server.env or with what has already been merged if
             mergedProperties = (Properties) combineServerEnvProperties(serverEnvProps, configuredServerEnvProps);
         }
 
-        if ((server.env != null && !server.env.isEmpty()) || !envProjectProps.isEmpty()) {
+        if (!configuredProps.isEmpty()) {
             if (serverEnvPath != null) {
-                logger.warn("The " + serverEnvPath + " file is merged with inlined configuration.")
+                logger.debug("The " + serverEnvPath + " file is merged with inlined configuration.")
+            }
+            else {
+                logger.debug("The " + getServerDir(project).getCanonicalPath() + " file is merged with inlined configuration.")
             }
 
             mergedProperties = combineServerEnvProperties(mergedProperties, configuredProps);
         }
 
         if(!mergedProperties.isEmpty()) {
             writeServerEnvProperties(envFile, mergedProperties);
+            return setServerEnvPathHelperForAppendServerEnv(envFile, configuredProps, serverEnvPath)
+        }
+
+        return serverEnvPath;
+    }
+
+    private String setServerEnvPathHelperForAppendServerEnv(File envFile, Properties configuredProps, String serverEnvPath) {
+        boolean configDirEnvMerged = envFile != null;
+        boolean serverEnvFileMerged = server.serverEnvFile != null && server.serverEnvFile.exists()
+        boolean inlineEnvPropsMerged = !configuredProps.isEmpty()
+
+        if(configDirEnvMerged && serverEnvFileMerged && inlineEnvPropsMerged) {
+            return "merging env properties, serverEnvFile" +  server.serverEnvFile.getCanonicalPath() + ", and configDir server.env" +  serverEnvPath
+        }
+        if (configDirEnvMerged && serverEnvFileMerged) {
+            return "merging serverEnvFile" +  server.serverEnvFile.getCanonicalPath() + ", and configDir server.env" +  serverEnvPath
+        }
+        if (configDirEnvMerged && inlineEnvPropsMerged) {
+            return "merging env properties and configDir server.env" + serverEnvPath 
+        }
+        if (serverEnvFileMerged && inlineEnvPropsMerged) {
+            return "merging env properties and serverEnvFile" +  server.serverEnvFile.getCanonicalPath() 
```

Comment:
There's actually more combinations than this I think. Perhaps we can use an approach to append to the String that gets returned. Start with "merging" and then for each boolean you check add " abc," then " cde," etc.
Once you've checked each boolean and added that part to the String, replace the last `,` with a `.` and return the String.

Reply:
Yeah, that's what I was thinking too. Can use a StringBuilder for this 👍